dashboard/cron.py

```diff
@@ -268,4 +81,339 @@ def util_function(sql_string, mysql_table, table_identifier=None):
     df.to_sql(con=engine, name=mysql_table, if_exists='append', index=False)
 
     # returns the row size of dataframe
-    return "inserted " + str(df.shape[0]) + " rows."
\ No newline at end of file
+    return  "inserted " + str(df.shape[0]) + " rows in table " + mysql_table + " for course " + UDW_course_id + ";"
+
+def update_cron_status(status):
+    logger.debug("in update_cron_status")
+
+    engine = create_engine("mysql+pymysql://{user}:{password}@{host}:{port}/{db}"
+                           .format(db = db_name,  # your mysql database name
+                                   user = db_user, # your mysql user for the database
+                                   password = db_password, # password for user
+                                   host = db_host,
+                                   port = db_port))
+    connection = engine.connect()
+    sql = "insert into cron_status (cron_date, status) values('" + str(datetime.datetime.now()) + "', '" + status + "')"
+    logger.debug("sql=" + sql)
+    connection.execute(sql)
+    connection.close()
+
+    logger.info("updated cron_status")
+
+# remove all records inside the specified table
+def deleteAllRecordInTable(tableName):
+    # delete all records in the table first
+    engine = create_engine("mysql+pymysql://{user}:{password}@{host}:{port}/{db}"
+                           .format(db = db_name,  # your mysql database name
+                                   user = db_user, # your mysql user for the database
+                                   password = db_password, # password for user
+                                   host = db_host,
+                                   port = db_port))
+    connection = engine.connect()
```

Comment:
[I saw that if you use the pattern
](https://docs.sqlalchemy.org/en/latest/core/connections.html#sqlalchemy.engine.Connection.detach)
```
with engine.connect() as con:
    conn.execute("SOME SQL") 
```

Then you don't have to close because connection is fully closed after that.